codec-http2/src/test/java/io/netty/handler/codec/http2/HpackDecoderTest.java

```diff
@@ -455,4 +455,28 @@ public void testDecodeLargerThanMaxHeaderListSizeButSmallerThanMaxHeaderListSize
             in.release();
         }
     }
+
+    @Test
+    public void testDecodeCountsNamesOnlyOnce() throws Http2Exception {
+        ByteBuf in = Unpooled.buffer(200);
+        try {
+            hpackDecoder.setMaxHeaderListSize(3500, 4000);
+            HpackEncoder hpackEncoder = new HpackEncoder(true);
+
+            // encode headers that are slightly larger than maxHeaderListSize
+            // but smaller than maxHeaderListSizeGoAway
+            Http2Headers toEncode = new DefaultHttp2Headers();
+            toEncode.add(String.format("%03000d", 0).replace('0', 'f'), "value");
+            toEncode.add("accept", "value");
+            hpackEncoder.encodeHeaders(1, in, toEncode, NEVER_SENSITIVE);
+
+            // decode the headers, we should get an exception, but
```

Comment:
this comment is a bit confusing... when we "get an exception" shouldn't we need to catch it ?

Reply:
oh, I just copy/pasted this from the previous test.  I'll just delete it.

Reply:
update, PTAL